File: Backend/src/main/java/com/tarikturkdil/dto/PinResponse.java
package com.tarikturkdil.dto;

import lombok.Data;

@Data
public class PinResponse extends DtoBase{
	private String title;
    private String description;
    private String imageUrl;
    private String ownerUsername;
    // İleride pinin sahibini de dönmek istersen UserResponse ekleyebilirsin
    private boolean likedByMe;
    private boolean savedByMe;
}

File: Frontend/app/src/main/java/com/tarikturkdil/photoProject/data/remote/dto/PinResponse.kt
package com.tarikturkdil.photoProject.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class PinResponse(
    val id: Long,
    val title: String,
    val description: String? = null,
    val imageUrl: String,
    val ownerUsername: String? = null,
    val likedByMe: Boolean = false,
    val savedByMe: Boolean = false,
    val createTime: String? = null
)